agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -4319,6 +4328,85 @@ bool backhaul_manager::add_link_metrics(const sMacAddr &reporter_al_mac,
 
     return true;
 }
+
+bool backhaul_manager::handle_channel_selection_request(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                        const std::string &src_mac)
+{
+    const auto mid = cmdu_rx.getMessageId();
+
+    LOG(DEBUG) << "Forwarding CHANNEL_SELECTION_REQUEST to son_slave, mid=" << std::hex << int(mid);
+    m_expected_channel_selection.mid = mid;
+    for (auto &socket : slaves_sockets) {
+        m_expected_channel_selection.requests.emplace_back(socket->radio_mac);
+    }
+    // Forward to all slaves
```

Comment:
Added advanced comment message in that place.

Reply:
> This is quite dangerous since we are handling this message by doing nothing and having faith that the default error handling will do the job for us. IIRC such default handling is going to change in a near future and this code will stop working.

Yes it will change, but in the best way possible: there will be no slaves any more, so no forwarding needs to be done at all.